File: Project Java & React/BackEnd/demo/src/main/java/com/app/services/FoodService.java
package com.app.services;

import java.util.List;
import java.util.Optional;

import org.springframework.http.ResponseEntity;

import com.app.Entities.Food;
import com.app.dto.FoodDto;

public interface FoodService {

	FoodDto addFood(FoodDto food/*, int catid*/);

	List<FoodDto> getAllFoods();

	String deleteFood(Integer food_id);

	Food addToCart(Integer food_id);

	FoodDto getbyId(int foodId);

	FoodDto updateFood(int foodId, FoodDto newfood);
	
//	List<FoodDto> getProductByCategory(int catid);



}

File: Project Java & React/FrontEnd/src/Components/Services/foodservice.js

import httpCommon from "../../http-common"

class foodservice {

    getAllFoods(){
        return httpCommon.get('/food/viewall');
    }

    addFood (food){
        return httpCommon.post('/food/addfood',food);
    }

    getFoodById(foodId){
        return httpCommon.get('/food/viewbyid/'+foodId)
    }

    updateFood(foodId , food){
        return httpCommon.put('/food/update/'+foodId , food)
    }

    deleteFood(foodId){
        return httpCommon.delete('/food/delete/'+foodId)
    }

}




// const add= (data) => {
//     return httpCommon.post('/addfood',data);
// };

// const getall = () => {
//     return httpCommon.get('/viewall');
// };

// const getFoodById = (foodId) => {
//     return httpCommon.get('/viewbyid',foodId);
// }


export default new foodservice();